src/main/java/com/nylas/NeuralMessageOptions.java

```diff
@@ -0,0 +1,34 @@
+package com.nylas;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class NeuralMessageOptions {
+	private final Boolean ignore_links;
+	private final Boolean ignore_images;
+	private final Boolean ignore_tables;
+	private final Boolean remove_conclusion_phrases;
+	private final Boolean images_as_markdowns;
+
+	public NeuralMessageOptions(Boolean ignore_links,
+								Boolean ignore_images,
+								Boolean ignore_tables,
+								Boolean remove_conclusion_phrases,
+								Boolean images_as_markdowns) {
+		this.ignore_links = ignore_links;
+		this.ignore_images = ignore_images;
+		this.ignore_tables = ignore_tables;
+		this.remove_conclusion_phrases = remove_conclusion_phrases;
+		this.images_as_markdowns = images_as_markdowns;
+	}
+
+	public Map<String, Boolean> toMap() {
+		Map<String, Boolean> map = new HashMap<>();
+		map.put("ignore_links", ignore_links);
+		map.put("ignore_images", ignore_images);
+		map.put("ignore_tables", ignore_tables);
+		map.put("remove_conclusion_phrases", remove_conclusion_phrases);
+		map.put("images_as_markdowns", images_as_markdowns);
```

Comment:
In the current implementation I didn't want to use primitive types because it would force the user to have to choose between true or false. Regarding having to check each field, we can do that but I tested the code as is and I'm pretty sure when the SDK converts the map to a JSON it ignores null fields. Example I've tested one with only `ignore_images` set and the rest were not (so, null) and the JSON that sent to the API only had `ignore_images` set.

Reply:
I'd prefer not to rely on the behavior for handling nulls if possible.

Reply:
Moved to the safer `Maps.putIfNotNull` when setting!